### PR TITLE
Add drone-based sandbox system

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Add a sandbox system that runs a drone sim from our [open-core](https://github.com/resim-ai/open-core).

## Guide to reproduce test results.
Build and push the new system images and experiences and make sure they work in the sanbox.

## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [ ] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: systems/drone/build.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+cp -Lr "${RESIM_RUN_RUNFILES_DIR}" bin/
+cp -Lr "${MAKE_VISUALIZATION_LOG_RUNFILES_DIR}" bin/
+
+# Copy all python libs over
+c=`cat <<EOF
+import shutil
+import os
+
+d = "${EXP_PROTO_DIR}"
+for root, dirs, files in os.walk(d):
```
Comment (iainjwhiteside) [inline systems/drone/build.sh:33]: 😬

File: systems/drone/build_and_push.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+docker push "${_FULL_METRICS_TAG}"
+
+echo "Registering build with ReSim..."
+resim metrics-builds create \
+      --url "${RESIM_API_URL}" \
+      --auth-url "${RESIM_AUTH_URL}" \
+      --name "A ReSim sandbox metrics build." \
+      --image "${_FULL_METRICS_TAG}" \
+      --project "${RESIM_SANDBOX_PROJECT}" \
+      --version "${RESIM_SANDBOX_BUILD_VERSION}"
```
Comment (iainjwhiteside) [inline systems/drone/build_and_push.sh:95]: `--systems ${RESIM_SANDBOX_SYSTEM}`

File: systems/drone/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+INPUTS="/tmp/resim/inputs"
+OUTPUTS="/tmp/resim/outputs"
+
+
+# Apply any parameters needed to the experience
+
+c=`cat <<EOF
```
Comment (iainjwhiteside) [inline systems/drone/entrypoint.sh:9]: it is a bit unfortunate that we mix the build/experience params concept here... nothing for it for now

File: systems/drone/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        exp = text_format.Parse(expfile.read(), Experience.Experience())
+    for movement_model in exp.dynamic_behavior.storyboard.movement_models:
+        if movement_model.HasField("ilqr_drone"):
+            movement_model.ilqr_drone.velocity_cost = eval(params["velocity_cost"])
+
+    with open("${INPUTS}/experience.sim", "w") as expfile:
+        expfile.write(str(exp))
+apply_params()
+EOF`
+PYTHONPATH=/resim python3 -c "$c"
```
Comment (iainjwhiteside) [inline systems/drone/entrypoint.sh:35]: any value in landing this as an actual python file?

**Review by iainjwhiteside — APPROVED** [review]
I'd be happy to land this as-is, though it might benefit from a readme to help navigate

File: systems/drone/build_and_push.sh (hunk 2 above)
Comment (mikebauer) [reply systems/drone/build_and_push.sh:95]: I don't think the CLI requires system. I just checked on the latest CLI.
```bash
$ resim metrics-build create
2024/05/21 17:12:36 required flag(s) "image", "name", "project", "version" not set
```

**Comment by mikebauer** [comment]
> I'd be happy to land this as-is, though it might benefit from a readme to help navigate

I made a ticket to track adding a README, but I'm gonna go ahead and land since I don't think I'll have time to work on it imminently.

**Pull request merged**